src/sentry/tasks/relay.py

```diff
@@ -275,6 +307,29 @@ def schedule_invalidate_project_config(
         },
     )
 
+    # Use the original arguments to this function to set the debounce key.
     projectconfig_debounce_cache.invalidation.debounce(
-        public_key=public_key, project_id=project_id, organization_id=organization_id
+        organization_id=organization_id, project_id=project_id, public_key=public_key
     )
+
+
+@instrumented_task(
+    name="sentry.tasks.relay.invalidate_all_project_configs",
+    queue="relay_config_bulk",
+    acks_late=True,
+    soft_time_limit=25 * 60,  # 25mins
+    time_limit=25 * 60 + 5,
+)
+def invalidate_all_project_configs(**kwargs):
```

Comment:
yeah, I considered this but really I don't think this is needed.  All tasks need to be executed as quickly as possible, we may as well pile them in.  With the improved deduplication behaviour introduced here I think this is rather good, because we're not stalling any tasks that need to be run.  If you batch you'll be merging in other invalidations for the projects not in the current batch, which is kind of a waste.

Plus you don't have to deal with a bunch of edge conditions around the batches and database entries still being the same.